tests/Unit/Evolution/DiscontinuousGalerkin/SlopeLimiters/Test_Hierarchical.cpp

```diff
@@ -0,0 +1,2383 @@
+  // We don't care about the ElementId for these tests, just the direction.
+  Limiter::PackagedData& package_data_upper = neighbor_data[std::make_pair(
+      Direction<dim>::upper_xi(), ElementId<dim>{0})];
+  Limiter::PackagedData& package_data_lower = neighbor_data[std::make_pair(
+      Direction<dim>::lower_xi(), ElementId<dim>{0})];
+
+  package_data_upper.volume_data.initialize(num_pts);
+  package_data_upper.mesh = mesh;
+  package_data_lower.volume_data.initialize(num_pts);
+  package_data_lower.mesh = mesh;
+
+  Scalar<DataVector> self_lnc(num_pts, 0.0);
+  tnsr::I<DataVector, dim> self_vector(num_pts, 0.0);
+  DataVector expected(num_pts);
+  const auto helper =
+      [
+        &element, &expected, &hierarchical, &mesh, &neighbor_data,
+        &package_data_lower, &package_data_upper, &self_lnc, &self_vector, &x
+      ](const ModalVector& upper_coeffs,
+        const std::vector<double>& initial_coeffs,
+        const ModalVector& lower_coeffs,
+        const std::vector<double>& expected_coeffs) noexcept {
+    set_legendre(&get(self_lnc), mesh, x, initial_coeffs);
+    get(get<::Tags::Modal<ScalarTag<0>>>(package_data_upper.volume_data)) =
+        upper_coeffs;
+    get(get<::Tags::Modal<ScalarTag<0>>>(package_data_lower.volume_data)) =
+        lower_coeffs;
+    for (size_t i = 0; i < dim; ++i) {
+      set_legendre(&self_vector.get(i), mesh, x, initial_coeffs);
+      get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_upper.volume_data)
+          .get(i) = upper_coeffs;
+      get<::Tags::Modal<VectorTag<dim, 0>>>(package_data_lower.volume_data)
+          .get(i) = lower_coeffs;
+    }
+    hierarchical(&self_lnc, &self_vector, element, mesh, neighbor_data);
+    set_legendre(&expected, mesh, x, expected_coeffs);
+    CHECK_ITERABLE_APPROX(get(self_lnc), expected);
+    for (size_t i = 0; i < dim; ++i) {
+      CHECK_ITERABLE_APPROX(self_vector.get(i), expected);
+    }
+  };
+  // Note: Throughout the tests 1.0e-18 is used as a +0.
+
+  // Limit no coefficients because the highest coefficient doesn't need
+  // limiting.
+  helper({7.0, 3.0, 4.1, 0.0}, {3.0, -2.0, 2.0, 1.0}, {1.0, -5.0, 1.0e-18, 0.0},
+         {3.0, -2.0, 2.0, 1.0});
+
+  // Limit the highest coefficient only:
+  // Limit top coeff from upper neighbor
+  helper({7.0, 3.0, 2.5, 0.0}, {3.0, -2.0, 2.0, 1.0}, {1.0, -5.0, 1.0e-18, 0.0},
+         {3.0, -2.0, 2.0, 0.5 * 0.99});
+  // Limit top coeff from lower neighbor
+  helper({7.0, 3.0, 4.1, 0.0}, {3.0, -2.0, 2.0, 1.0}, {1.0, -5.0, 1.3, 0.0},
+         {3.0, -2.0, 2.0, 0.7 * 0.99});
+  // Zero top coeff from upper neighbor
+  helper({7.0, 3.0, 1.9, 0.0}, {3.0, -2.0, 2.0, 1.0}, {1.0, -5.0, 1.0e-18, 0.0},
+         {3.0, -2.0, 2.0, 0.0});
+  // Zero top coeff from lower neighbor
+  helper({7.0, 3.0, 4.1, 0.0}, {3.0, -2.0, 2.0, 1.0}, {1.0, -5.0, 5.3, 0.0},
+         {3.0, -2.0, 2.0, 0.0});
+
+  // Limit the top 2 coeffs:
+  // Limit top upper neighbor, limit second upper neighbor
```

Comment:
When getting to second coeff, no need to explain how the threshold is obtained, but it would be nice to state its value anyway.

Reply:
Sorry, I'm not understanding what you're asking for here. Could you rephrase and/or give an example?